apps/whale-api/src/module.api/poolswap.pathfinding.service.ts

```diff
@@ -135,6 +135,11 @@ export class PoolSwapPathFindingService {
     const poolPairPaths: SwapPathPoolPair[][] = []
 
     for (const path of allSimplePaths(this.tokenGraph, fromTokenId, toTokenId)) {
+      // Max 3 hops, e.g. A --> B --> C --> D
+      if (path.length > 4) {
```

Comment:
heyy.. need clarify.. so the the `I-J` won't be in the path right??
```
I-J -> J-L -> L-M -> M-N
// to
I-J -> path(J-L -> L-M -> M-N)
```

Reply:
Ah in this example, `I-J -> J-L -> L-M -> M-N` is already 4 hops
path.length = `[I, J, L, M, N]`. Hence path.length > 4 will be ignored